test/integration/regression.spec.js

```diff
@@ -54,17 +55,175 @@ describe('regressions', function () {
     });
   });
 
-  describe('issue-2286: after doesn\'t execute if test was skipped in beforeEach', function () {
-    var afterWasRun = false;
-    describe('suite with skipped test for meta test', function () {
-      beforeEach(function () { this.skip(); });
-      after(function () { afterWasRun = true; });
-      it('should be pending', function () {});
-    });
-    after('meta test', function () {
-      afterWasRun.should.be.ok();
+  describe(
+    "issue-2286: after doesn't execute if test was skipped in beforeEach",
+    function () {
+      /**
+       * Generates tests for behavior of `this.skip()`
+       * @param {string} name - Name of Runnable abort via `this.skip()`
+       * @param {Array.<boolean>} expected - For each of the Runnable types,
+       *   whether or not the Runnable should have been run.  The order is
+       *   "beforeAll" x 2, "beforeEach" x 2, "test" x 2, "afterEach" x 2, then
+       *   "afterAll" x 2.  There should be 10 items in this array.
+       * @param {string} [mode=always] - One of 'always' or 'once'
+       */
+      function testPendingRunnables (name, expected, mode) {
+        mode = mode || 'always';
+        var spies = [];
+
+        function spy (skip) {
+          function wrapped () {
+            if ((!wrapped.runCount++ || mode === 'always') && skip) {
+              this.skip();
+            }
+          }
+
+          wrapped.runCount = 0;
+          spies.push(wrapped);
+          return wrapped;
+        }
+
+        describe(name, function () {
+          describe('meta', function () {
+            before(spy(name === 'beforeAll'));
+            before(spy(name === 'beforeAll'));
+            beforeEach(spy(name === 'beforeEach'));
+            beforeEach(spy(name === 'beforeEach'));
+            it('might be pending', spy(name === 'test'));
+            it('might be pending', spy(name === 'test'));
+            afterEach(spy(name === 'afterEach'));
+            afterEach(spy(name === 'afterEach'));
+            after(spy(name === 'afterAll'));
+            after(spy(name === 'afterAll'));
+          });
+
+          after(name + ' - ' + mode, function () {
+            map(spies, function (spy) {
+              return Boolean(spy.runCount);
+            })
+              .should
+              .deepEqual(expected);
+          });
+        });
+      }
+
+      testPendingRunnables('beforeAll', [
+        true, // beforeAll
```

Comment:
Booleans that map to calls are not intuitive. What if you used strings instead?

Possibly use `beforeAll` and `beforeAll-called-skip` (or something similar) to denote the state?

Then you wouldn't need comments to explain what's everything means.